crates/mdk-core/src/key_packages.rs

```diff
@@ -2029,4 +2109,333 @@ mod tests {
         // protocol level, but the current test validates the extension is present and basic
         // lifecycle works correctly.
     }
+
+    #[test]
+    fn test_key_package_base64_encoding() {
+        let config = crate::MdkConfig {
+            use_base64_encoding: true,
+        };
+
+        let mdk = crate::tests::create_test_mdk_with_config(config);
+        let test_pubkey =
+            PublicKey::from_hex("884704bd421671e01c13f854d2ce23ce2a5bfe9562f4f297ad2bc921ba30c3a6")
+                .unwrap();
+        let relays = vec![RelayUrl::parse("wss://relay.example.com").unwrap()];
+
+        let (key_package_str, tags) = mdk
+            .create_key_package_for_event(&test_pubkey, relays)
+            .expect("Failed to create key package");
+
+        // Verify the content is valid base64 (no prefix in tag-based approach)
+        assert!(
+            BASE64.decode(&key_package_str).is_ok(),
+            "Content should be valid base64, got: {}",
+            key_package_str
+        );
+
+        // Verify the encoding tag is present
+        let encoding_tag = tags
+            .iter()
+            .find(|t| t.as_slice().first() == Some(&"encoding".to_string()));
+        assert!(
+            encoding_tag.is_some(),
+            "Should have encoding tag when using base64"
+        );
+        assert_eq!(
+            encoding_tag.unwrap().as_slice().get(1).map(|s| s.as_str()),
+            Some("base64"),
+            "Encoding tag should be 'base64'"
+        );
+
+        // Verify we can parse it back with base64 encoding
+        let parsed = mdk
+            .parse_serialized_key_package(&key_package_str, ContentEncoding::Base64)
+            .expect("Failed to parse base64 key package");
+        assert_eq!(parsed.ciphersuite(), DEFAULT_CIPHERSUITE);
+    }
+
+    #[test]
+    fn test_key_package_hex_encoding_legacy() {
+        let config = crate::MdkConfig {
+            use_base64_encoding: false,
+        };
+
+        let mdk = crate::tests::create_test_mdk_with_config(config);
+        let test_pubkey =
+            PublicKey::from_hex("884704bd421671e01c13f854d2ce23ce2a5bfe9562f4f297ad2bc921ba30c3a6")
+                .unwrap();
+        let relays = vec![RelayUrl::parse("wss://relay.example.com").unwrap()];
+
+        let (key_package_str, tags) = mdk
+            .create_key_package_for_event(&test_pubkey, relays)
+            .expect("Failed to create key package");
+
+        // Verify it's valid hex
+        assert!(hex::decode(&key_package_str).is_ok(), "Should be valid hex");
+
+        // Verify no encoding tag is present (hex is the default)
+        let encoding_tag = tags
+            .iter()
+            .find(|t| t.as_slice().first() == Some(&"encoding".to_string()));
+        assert!(
+            encoding_tag.is_none(),
+            "Should NOT have encoding tag when using hex (default)"
+        );
```

Comment:
Related to the prior comments, I think we should still include the tag (and verify that we're doing so here in the tests)